File: codingTest/programmers/python/위장.py
def solution(cloths):
    hash_map = {}
    count = 1
    for item in cloths:
        if item[1] in hash_map: hash_map[item[1]] += 1
        else: hash_map[item[1]] = 1
    for item in hash_map: # 선택받지 못하는 경우 (+1)
        hash_map[item] += 1
    for value in hash_map.values():
        count *= value
    return count - 1

File: codingTest/programmers/javascript/위장.js
function solution(cloths) {
  const hashMap = {};
  for (let item of cloths) hashMap[item[1]] = (hashMap[item[1]] || 0) + 1;
  for (let key in hashMap) hashMap[key]++;
  let count = 1;
  Object.values(hashMap).forEach(e => (count *= e));
  return count - 1;
}
